src/shogun/optimization/GradientDescendUpdater.h

```diff
@@ -95,6 +95,11 @@ class GradientDescendUpdater: public DescendUpdaterWithCorrection
 	virtual void update_variable(SGVector<float64_t> variable_reference,
 		SGVector<float64_t> raw_negative_descend_direction);
 
+
+	virtual LearningRate* get_learning_rate()
+	{
+		return m_learning_rate;
```

Comment:
Once we use C++11 compiler, we can use `std::shared_ptr`.
Last time, the buildbot (travis) complained about the `auto` keyword.
So, for now I use raw pointer and users have to new/delete pointers in the `main`.